Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -1822,18 +1842,46 @@ private static void ReportFileAccesses(ExecutionResult processExecutionResult, F
             int numPathSetsDownloaded = 0, numCacheEntriesVisited = 0;
 
             using (operationContext.StartOperation(PipExecutorCounter.CheckProcessRunnableFromCacheDuration))
+            using (var strongFingerprintComputationListWrapper = SchedulerPools.StrongFingerprintDataListPool.GetInstance())
+            {
+                List<BoxRef<ProcessStrongFingerprintComputationData>> strongFingerprintComputationList =
+                    strongFingerprintComputationListWrapper.Instance;
+
+                var result = await innerComputeAsync(strongFingerprintComputationList);
+
+                // Update the strong fingerprint computations list
+                processFingerprintComputationResult.StrongFingerprintComputations = strongFingerprintComputationList.SelectArray(s => s.Value);
+
+                using (operationContext.StartOperation(PipExecutorCounter.CheckProcessRunnableFromCacheExecutionLogDuration))
+                {
+                    // TODO: How to log to execution log
+                    environment.State.ExecutionLog?.ProcessFingerprintComputation(processFingerprintComputationResult);
+                }
+
+                return result;
+            }
+
+            // Extracted local function with main logic for performing the cache lookup.
+            // This is done to ensure that execution log logging is always done even in cases of early return (namely augmented weak fingerprint cache lookup
+            // defers to an inner cache lookup and performs an early return of the result)
+            async Task<RunnableFromCacheResult> innerComputeAsync(List<BoxRef<ProcessStrongFingerprintComputationData>> strongFingerprintComputationList)

```

Comment:
>innerComputeAsync [](start = 48, length = 17)

nit: fix capitalization

Reply:
nm, it's local function... Can we not make it local?

---
In reply to: [322946518](https://github.com/microsoft/BuildXL/pull/857#discussion_r322946518) [](ancestors = 322946518)